nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/controller/ControllerFacade.java

```diff
@@ -564,6 +577,62 @@ private boolean implementsServiceType(final Class serviceType, final Class type)
         return dtoFactory.fromDocumentedTypes(getExtensionManager().getExtensions(ReportingTask.class), bundleGroupFilter, bundleArtifactFilter, typeFilter);
     }
 
+
+    /**
+     * Gets the RuntimeManifest for this overall NiFi instance.
+     *
+     * @return the runtime manifest
+     */
+    public RuntimeManifest getRuntimeManifest() {
+        final ExtensionManager extensionManager = getExtensionManager();
+        final Set<Bundle> allBundles = extensionManager.getAllBundles();
+
+        final Bundle frameworkBundle = NarClassLoadersHolder.getInstance().getFrameworkBundle();
+        final BundleDetails frameworkDetails = frameworkBundle.getBundleDetails();
+        final Date frameworkBuildDate = frameworkDetails.getBuildTimestampDate();
+
+        final BuildInfo buildInfo = new BuildInfo();
+        buildInfo.setVersion(frameworkDetails.getCoordinate().getVersion());
+        buildInfo.setRevision(frameworkDetails.getBuildRevision());
+        buildInfo.setCompiler(frameworkDetails.getBuildJdk());
+        buildInfo.setTimestamp(frameworkBuildDate == null ? null : frameworkBuildDate.getTime());
+
+        final RuntimeManifestBuilder manifestBuilder = new StandardRuntimeManifestBuilder()
+                .identifier("nifi")
+                .runtimeType("nifi")
```

Comment:
Is the value of `nifi` important for any other consuming service?  What do you think about promoting the value to a static variable?

Reply:
It is not really that important for a consumer, mostly just metadata information, but I will promote it to a constant